<a href="https://colab.research.google.com/github/GiovannaRochaDSM/Web_Scraping_Mineracao_Dados/blob/master/Web_Scraping_Giovanna_Rocha.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Atividade de Web Scraping - 19/10/2024**

#### **Giovanna da Rocha Machado - RA: 1131392213024**

---------------------------------


URL do site - Books (Romance)
- https://books.toscrape.com/catalogue/category/books/romance_8/index.html



A partir do web scrapping do site acima, vamos obter os dados abaixo:

**1.** Preço médio dos livros: Mostra o preço médio de todos os livros listados.

**2.** Contagem de livros disponíveis: Informa quantos livros estão disponíveis no estoque.

**3.** Livros mais caros: Exibe os 3 livros com os preços mais altos na lista


Vamos começar!

In [ ]:
# Importação de libs
import requests
from bs4 import BeautifulSoup

In [ ]:
# URL do site e headers do navegador
url = 'https://books.toscrape.com/catalogue/category/books/romance_8/index.html'
headers = {"User-Agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 16_6 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.6 Mobile/15E148 Safari/604.1"}

In [ ]:
# Requisição para a URL
response = requests.get(url, headers=headers)

In [ ]:
# Interpretar o HTML com BeautifulSoup
soup = BeautifulSoup(response.text, 'html.parser')

In [ ]:
# Localizar todas arcticles com a classe 'product_pod'
produtos = soup.find_all('article', class_='product_pod')

In [ ]:
precos = []
disponibilidade = 0

for produto in produtos:
    preco_str = produto.find('p', class_='price_color').text.strip()  # Removendo espaços em branco
    preco = preco_str[1:]  # Removendo símbolo da moeda, primeiro caractere
    preco = ''.join(filter(lambda x: x.isdigit() or x == '.', preco)) # Removendo caracteres não numéricos

    precos.append(float(preco))
    disponibilidade += 1

In [ ]:
# 1. Preço médio dos livros
preco_medio = sum(precos) / len(precos)

In [ ]:
# 2. Contagem de livros disponíveis
contagem_livros = disponibilidade

In [ ]:
# 3. Livros mais caros
livros_com_preco = [(produto.h3.a['title'], preco) for produto, preco in zip(produtos, precos)]
livros_mais_caros = sorted(livros_com_preco, key=lambda x: x[1], reverse=True)[:3]

In [ ]:
print(f'Preço médio dos livros: £{preco_medio:.2f}')
print()
print(f'Contagem de livros disponíveis: {contagem_livros}')
print()
print('3 livros mais caros:')
for titulo, preco in livros_mais_caros:
    print(f'Título: {titulo}, Preço: £{preco:.2f}')


Preço médio dos livros: £31.04

Contagem de livros disponíveis: 20

3 livros mais caros:
Título: The Perfect Play (Play by Play #1), Preço: £59.99
Título: Suddenly in Love (Lake Haven #1), Preço: £55.99
Título: Take Me Home Tonight (Rock Star Romance #3), Preço: £53.98


Após realizar o web scraping do site, obtivemos as seguintes informações:

1. Preço médio dos livros: O preço médio de todos os livros listados é de `£59.99.`

2. Contagem de livros disponíveis: No total, encontramos 20 livros disponíveis em estoque nesta página.

3. Livros mais caros: Os três livros com os preços mais altos são:

Título: "The Perfect Play", Preço: `£59.99.`

Título: "Suddenly in Love", Preço: `£55.99.`

Título: "Take me Home Tonight", Preço: `£53.98.`
